src/main/java/com/amazon/dlic/auth/ldap2/LDAPAuthorizationBackend2.java

```diff
@@ -376,10 +380,12 @@ private void fillRoles0(final User user, final AuthCredentials optionalAuthCreds
                 for (final LdapName roleLdapName : nestedReturn) {
                     final String role = getRoleFromEntry(connection, roleLdapName, roleName);
 
-                    if (!Strings.isNullOrEmpty(role)) {
-                        user.addRole(role);
+                    if (excludeRolesMatcher.test(role)) {
+                        if (isDebugEnabled) {
+                            log.debug("Role was excluded or empty attribute '{}' for entry {}", roleName, roleLdapName);
```

Comment:
FYI - To validate logging statements, edit  `src/integrationTest/resources/log4j2-test.properties` to enable debug logging for testing and then you can validate the log entries with a LogsRule, pretty fancy.

```java
@Rule
public LogsRule logsRule = new LogsRule("com.amazon.dlic.auth.http.jwt.HTTPJwtAuthenticator");
```
```java
logsRule.assertThatContainExactly("Invalid or expired JWT token.");
```